### ISSUE TITLE
Design consistent UI interface for keyboard+mouse and touch devices

### ISSUE THREAD

**Issue by brprice** [hackworthltd/primer-app#188]
We should come up with some system of input controls (click, right-click, keypresses, hover, dragging etc) for our webapp to use, bearing in mind that it should work smoothly for both traditional desktops and also mobile devices. It may be that we want different controls on different devices (e.g. right-click on an actual mouse is an easy gesture, but does not exist as such on touchscreens), though it would be nice if the correspondence was obvious.

There are also potential accessibility concerns here. For example, small targets, swift double-click-ing...

Currently we have the "interactive tree" story in our storybook #167  (which is intended for visualisation during development and has not worried about touch devices). It uses `onclick` to grow a tree and `oncontextmenu` ("right-click") to prune. This is nice with a traditional mouse; it works on android (long-press triggers pruning); but does not work on iPad (I cannot trigger pruning; I don't know how easy it would be to "fix" - I have not investigated much). Whilst ok for development visualisation, it is clearly not acceptable in a released product!

The input actions I think we will certainly want are (in terms of traditional desktop input):
- "click" to select a node or press a button
- "type" to fill in free-form text boxes
- "click+drag" to pan a viewport

I don't know whether we will want anything like the following:
- "scroll wheel" to zoom
- "right-click"/"middle-click"/"double-click" to trigger additional actions (perhaps a shortcut for "do the obvious thing to this node")
- "hover" to pop up a tooltip / info box

Of course, there may be other things I have not thought of here!
